### PR TITLE
Feature/validators tests

### PR DESCRIPTION
**Pull request by BoroviyOrest**
## JIRA

* [Main JIRA ticket](http://ssu-jira.softserveinc.com/browse/PYTHON-274)

## Code reviewers

- [ ] @meelros
- [ ] @Kardan1123
- [ ] @mizin4ik
- [ ] @dimigor
- [ ] @PetrushynskyiOleksii
- [ ] @Sasha1152

### Second Level Review

- [ ] @lhalam

## Summary of change

Add tests for validators and fix several validators

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: way_to_home/tests/unittests/utils/testvalidators.py [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+                     'key2': 'test'}
+        test_required_keys = ['key1', 'key2']
+        result = none_validator_for_required_keys(data=test_data, keys_required=test_required_keys)
+        self.assertTrue(result)
+
+    def test_date_validator(self):
+        """Method that tests the date_validator function"""
+        invalid_date = 123
+        result = date_validator(date=invalid_date)
+        self.assertEqual(result, None)
```
Comment (PetrushynskyiOleksii) [inline way_to_home/tests/unittests/utils/testvalidators.py]:
```suggestion
        self.assertEqual(result, None)
```
You can use `assertIsNone` in this case.

File: way_to_home/tests/unittests/utils/testvalidators.py [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+
+        invalid_date = '12323fd'
+        result = date_validator(date=invalid_date)
+        self.assertFalse(result)
```
Comment (PetrushynskyiOleksii) [inline way_to_home/tests/unittests/utils/testvalidators.py]:
```suggestion
        self.assertFalse(result)
```
I think you should change it to `assertIsNone` because it returns `None`
```
>>> result = date_validator('12323fd')
>>> print(result)
None
```

File: way_to_home/tests/unittests/utils/testvalidators.py [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+        """Method that tests the week_day_validator function"""
+        invalid_week_day = 'fgs'
+        result = week_day_validator(week_day=invalid_week_day)
+        self.assertFalse(result)
+
+        invalid_week_day = 15
+        result = week_day_validator(week_day=invalid_week_day)
+        self.assertFalse(result)
+
+        invalid_week_day = -15
```
Comment (PetrushynskyiOleksii) [inline way_to_home/tests/unittests/utils/testvalidators.py]: I think it is better to display week days that are closest to valid: `-1` and `7`.

File: way_to_home/tests/unittests/utils/testvalidators.py [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+        result = week_day_validator(week_day=invalid_week_day)
+        self.assertFalse(result)
+
+        test_week_day = 5
+        result = week_day_validator(week_day=test_week_day)
+        self.assertTrue(result)
```
Comment (PetrushynskyiOleksii) [inline way_to_home/tests/unittests/utils/testvalidators.py:337]: I think it is better to use for loop in range (0, 6) to validate all suitable week days.

File: way_to_home/tests/unittests/utils/testvalidators.py [hunk 5]
```diff
@@ -0,0 +1,265 @@
...
+
+        invalid_data = {'email': 123,
+                        'password': None}
+        result = credentials_validator(data=invalid_data)
+        self.assertFalse(result)
+
+        invalid_data = {'email': 123,
+                        'password': 123}
+        result = credentials_validator(data=invalid_data, update=True)
+        self.assertFalse(result)
```
Comment (PetrushynskyiOleksii) [inline way_to_home/tests/unittests/utils/testvalidators.py:381]: With `update=True` you just need to display tests when one of the fields will be `None` and one of the fields will not exist in the input dictionary.

**Review by PetrushynskyiOleksii — CHANGES_REQUESTED** [review]

**Review by dimigor — DISMISSED** [review]

**Review by PetrushynskyiOleksii — DISMISSED** [review]

**Review by PetrushynskyiOleksii — APPROVED** [review]

**Review by dimigor — APPROVED** [review]
